src/index.js

```diff
@@ -486,23 +578,31 @@ class JanusAdapter {
   }
 
   sendData(clientId, dataType, data) {
-    this.publisher.unreliableChannel.send(
-      JSON.stringify({ clientId, dataType, data })
-    );
+    if (this.publisher) {
+      this.publisher.unreliableChannel.send(
+        JSON.stringify({ clientId, dataType, data })
+      );
+    }
   }
 
   sendDataGuaranteed(clientId, dataType, data) {
-    this.publisher.reliableChannel.send(
-      JSON.stringify({ clientId, dataType, data })
-    );
+    if (this.publisher) {
+      this.publisher.reliableChannel.send(
+        JSON.stringify({ clientId, dataType, data })
+      );
+    }
   }
 
   broadcastData(dataType, data) {
-    this.publisher.unreliableChannel.send(JSON.stringify({ dataType, data }));
+    if (this.publisher) {
+      this.publisher.unreliableChannel.send(JSON.stringify({ dataType, data }));
+    }
   }
 
   broadcastDataGuaranteed(dataType, data) {
-    this.publisher.reliableChannel.send(JSON.stringify({ dataType, data }));
+    if (this.publisher) {
+      this.publisher.reliableChannel.send(JSON.stringify({ dataType, data }));
+    }
```

Comment:
Hmm, it's a bit troubling if something tries to send data before a publisher is established. Maybe add a console warning in an else statement in these methods?

Reply:
Yeah, I'll add that. I don't like that we have to do this check at all. It feels bad that a call to `broadcastDataGuaranteed` is not guaranteed to broadcast data or error. Unfortunately I'm not sure how to fix this without making larger changes to networked-aframe to add the concept of an adapter that is in a reconnecting state and avoid sending data altogether. I don't think I want to tackle that just yet.